eth_dummy_phy/src/esp_eth_phy_dummy.c

```diff
@@ -0,0 +1,188 @@
+
+        bool peer_pause_ability = false; // always false
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_PAUSE, (void *)peer_pause_ability), err, TAG, "change pause ability failed");
+
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_LINK, (void *)dummy_phy->link), err, TAG, "change link failed");
+    }
+
+err:
+    return ret;
+}
+
+static esp_err_t set_link(esp_eth_phy_t *phy, eth_link_t link)
+{
+    esp_err_t ret = ESP_OK;
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    esp_eth_mediator_t *eth   = dummy_phy->eth;
+
+    if (dummy_phy->link != link) {
+        dummy_phy->link = link;
+        // link status changed, inmiedately report to upper layers
+        ESP_GOTO_ON_ERROR(eth->on_state_changed(eth, ETH_STATE_LINK, (void *)dummy_phy->link), err, TAG, "change link failed");
+    }
+err:
+    return ret;
+}
+
+static esp_err_t set_mediator(esp_eth_phy_t *phy, esp_eth_mediator_t *eth)
+{
+    esp_err_t ret = ESP_OK;
+    ESP_GOTO_ON_FALSE(eth, ESP_ERR_INVALID_ARG, err, TAG, "mediator can't be null");
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    dummy_phy->eth = eth;
+err:
+    return ret;
+}
+
+static esp_err_t reset_hw(esp_eth_phy_t *phy)
+{
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+
+    if (dummy_phy->reset_gpio_num >= 0) {
+        esp_rom_gpio_pad_select_gpio(dummy_phy->reset_gpio_num);
+        gpio_set_direction(dummy_phy->reset_gpio_num, GPIO_MODE_OUTPUT);
+        gpio_set_level(dummy_phy->reset_gpio_num, 0);
+        esp_rom_delay_us(100);
+        gpio_set_level(dummy_phy->reset_gpio_num, 1);
+    }
+    return ESP_OK;
+}
+
+static esp_err_t autonego_ctrl(esp_eth_phy_t *phy, eth_phy_autoneg_cmd_t cmd, bool *autonego_en_stat)
+{
+    switch (cmd) {
+    case ESP_ETH_PHY_AUTONEGO_RESTART:
+    // Fallthrough
+    case ESP_ETH_PHY_AUTONEGO_EN:
+    // Fallthrough
+    case ESP_ETH_PHY_AUTONEGO_DIS:
+        return ESP_ERR_NOT_SUPPORTED; // no autonegotiation operations are supported
+    case ESP_ETH_PHY_AUTONEGO_G_STAT:
+        // report that auto-negotiation is not supported
+        *autonego_en_stat = false;
+        break;
+    default:
+        return ESP_ERR_INVALID_ARG;
+    }
+    return ESP_OK;
+}
+
+static esp_err_t set_speed(esp_eth_phy_t *phy, eth_speed_t speed)
+{
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    dummy_phy->link = ETH_LINK_DOWN;
+    dummy_phy->speed = speed;
+    get_link(phy); // propagate the change to higher layer
+    return ESP_OK;
+}
+
+static esp_err_t set_duplex(esp_eth_phy_t *phy, eth_duplex_t duplex)
+{
+    phy_dummy_t *dummy_phy = __containerof(phy, phy_dummy_t, parent);
+    dummy_phy->link = ETH_LINK_DOWN;
+    dummy_phy->duplex = duplex;
+    get_link(phy); // propagate the change to higher layer
+    return ESP_OK;
+}
+
+static esp_err_t do_nothing(esp_eth_phy_t *phy)
+{
+    return ESP_OK;
```

Comment:
question, do you prefer to return ESP_OK or ESP_ERR_NOT_SUPPORTED

Reply:
I prefer to use ESP_OK, since goal of all these functions is to do nothing and we don't want to initiate any error handling at upper level.